src/components/DxpFacilitySwitcher.vue

```diff
@@ -0,0 +1,164 @@
+          </div>
+          <div v-else>
+            <ion-item v-for="facility in filteredFacilities" :key="facility.facilityId">
+              <ion-radio label-placement="end" justify="start" :value="facility.facilityId">
+                <ion-label>
+                  {{ facility.facilityName }}
+                  <p>{{ facility.facilityId }}</p>
+                </ion-label>
+              </ion-radio>
+            </ion-item>
+          </div>
+        </ion-list>
+      </ion-radio-group>
+
+      <ion-fab vertical="bottom" horizontal="end" slot="fixed">
+        <ion-fab-button :disabled="selectedFacilityId === currentFacility.facilityId" @click="setFacility">
+          <ion-icon :icon="saveOutline" />
+        </ion-fab-button>
+      </ion-fab>
+    </ion-content>
+  </ion-modal>
+</template>
+
+<script setup lang="ts">
+import { 
+  IonButton,
+  IonButtons,
+  IonCard,
+  IonCardContent,
+  IonCardHeader,
+  IonCardTitle,
+  IonContent,
+  IonFab,
+  IonFabButton,
+  IonHeader,
+  IonIcon,
+  IonItem,
+  IonLabel,
+  IonList,
+  IonModal,
+  IonRadio,
+  IonRadioGroup,
+  IonSearchbar,
+  IonSpinner,
+  IonTitle,
+  IonToolbar
+} from '@ionic/vue';
+import { closeOutline, saveOutline } from "ionicons/icons";
+import { useUserStore } from 'src/store/user';
+import { computed, ref } from 'vue';
+
+const userStore = useUserStore();
+
+const facilities = computed(() => userStore.getFacilites)
+const currentFacility = computed(() => userStore.getCurrentFacility)
+
+const facilityModal = ref()
+const queryString = ref('')
+const isLoading = ref(true);
+const filteredFacilities = ref([])
+const selectedFacilityId = ref(currentFacility.value.facilityId)
+
+const emit = defineEmits(["updateFacility"])
+
+const closeModal = () => {
+  facilityModal.value.$el.dismiss(null, 'cancel');
+}
+
+function loadFacilities() {
+  filteredFacilities.value = facilities.value;
+  isLoading.value = false;
+}
+
+const findFacility = () => {
+  isLoading.value = true
+  const searchedString = queryString.value.trim().toLowerCase();
+  if(searchedString) {
+    filteredFacilities.value = facilities.value.filter((facility: any) => 
+      facility.facilityName.toLowerCase().includes(searchedString) || 
+      facility.facilityId.toLowerCase().includes(searchedString)
+    );
+  } else {
+    filteredFacilities.value = facilities.value;
+  }
+  isLoading.value = false
+}
+
+async function selectSearchBarText(event: any) {
+  const element = await event.target.getInputElement()
+  element.select();
+}
+
+function preventSpecialCharacters($event: any) {
+  // Searching special characters fails the API, hence, they must be omitted
+  if(/[`!@#$%^&*()_+\-=\\|,.<>?~]/.test($event.key)) $event.preventDefault();
+}
+
+function setFacility() {
+  const selectedFacility = facilities.value.find((facility: any) => facility.facilityId === selectedFacilityId.value)
+  const isUpdated = userStore.setFacility(selectedFacility)
+  if(isUpdated) {
```

Comment:
We should not wait for whether the preference for user is set or not, instead we should allow the user to update the facility in the app, irrespective of preference set on the server.